src/components/header/_header.scss

```diff
@@ -14,11 +14,31 @@ $govuk-header-link-underline-thickness: 3px;
     float: right;
     margin-bottom: 10px;
 
+    @include govuk-media-query($from: desktop) {
+      padding-top: 10px;
+
+      .govuk-template--rebranded & {
+        padding-top: 26px;
+      }
+    }
+
     .govuk-template--rebranded & {
       margin-bottom: 15px;
     }
   }
 
+  .govuk-header__navigation + .hmrc-sign-out-nav {
+    padding-top: 4px;
+
+    @include govuk-media-query($from: tablet) {
+      padding-top: 10px;
+    }
+
+    @include govuk-media-query($from: desktop) {
+      padding-top: 0;
+    }
+  }
+
```

Comment:
switched away from has() to make it easier to understand since it's equivalent

and synced the property to use padding-top instead of margin, no issues with collapsing margins spotted